force-app/main/default/classes/CON_ContactMerge_CTRL.cls

```diff
@@ -950,98 +950,137 @@ public with sharing class CON_ContactMerge_CTRL {
+        allContacts.add((Contact)winningContact);
+        allContacts.addAll(losingContacts);
+
+        List<Contact> lockedContacts = lockAllContacts(allContacts);
+
+        if (lockedContacts == null || lockedContacts.size() != allContacts.size()) {
+            ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.Error, 
+                label.npe01.Contact_Merge_Error_Lock_failed  + ' ' + 
+                label.npe01.Contact_Merge_Error_please_retry) );
+            return null;
+        }
+
+        System.SavePoint sp = Database.setSavepoint();
+        try {
+            // We need to turn off Affiliation processing or there will be duplicate affiliates
+            // created after the merge.  As merge transactions do not fire child triggers
+            TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.flag.AFFL_doNotRunForContacts, true);
+            updateAndHandleWinningContact(winningContact);
+            TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.flag.AFFL_doNotRunForContacts, false);
+            
+            CON_ContactMerge.mergeContacts(winningContact, losingContacts);
+
+            winningContact = populateContactWithOnlyUniqueFields(winningContact.Id);
+            updateAndHandleWinningContact(winningContact);
+
+        } catch (Exception e) {
+            Database.rollback(sp);
+            ERR_Handler.processError(e, ERR_Handler_API.Context.CON);
+            ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.Error, label.npe01.Contact_Merge_Error_Merge_Failed + ' Error: ' + e));
+
+            return null;
+        }
+
+        return new PageReference('/' + winningContact.id);
+    }
+
+    // Find the winningContact record based the selected value of the winningContact FieldRow
+    private Contact getWinningContact() {
         for (FieldRow row : fieldRows) {
             if (row.showRadio && row.fieldName.equals(MASTER_KEY)) {
-                master = new Contact(id = row.selectedValue);
-                break;
+                return new Contact(Id = row.selectedValue);
             }
         }
+        
+        return null;
+    }
 
-        if (master != null) {
-            // Update the field values of master record based on the selected value for each field.
-            for (FieldRow row : fieldRows) {
-                if (row.showRadio && 
-                    !row.fieldName.equals(MASTER_KEY) && 
-                    row.selectedValue != master.id) {
-                    SObject selectedRecord = this.selectedRecords.get(row.selectedValue);
-
-                    // Sobject.put is not happy when sobject.get returns null. It throws an 
-                    // exception System.SObjectException: Illegal assignment from Object to String.
-                    // But instead when you pass a null directly to put, it works fine. 
-                    // And hence, this if statement.
-                    if (selectedRecord.get(row.fieldName) == null) {
```

Comment:
Are we potentially introducing a regression by changing this?

Reply:
I dont think so, I was testing this with apex anonymous and it seem like the put can accept null value now. I can roll back this change if needed.

Reply:
I'll do some testing and let you know!